sql: TestMultiTenantAdminFunction failed

sql.TestMultiTenantAdminFunction [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7896301?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7896301?buildTab=artifacts#/) on master @ [3910ac409195ab6ab2ec3e9f6eda553048afa23f](https://github.com/cockroachdb/cockroach/commits/3910ac409195ab6ab2ec3e9f6eda553048afa23f):


```
=== RUN   TestMultiTenantAdminFunction
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/b287de7194e17a0386ee1775f0d198ba/logTestMultiTenantAdminFunction795374909
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestMultiTenantAdminFunction
    multitenant_admin_function_test.go:346: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/b287de7194e17a0386ee1775f0d198ba/logTestMultiTenantAdminFunction795374909
--- FAIL: TestMultiTenantAdminFunction (142.16s)
=== RUN   TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck
    multitenant_admin_function_test.go:87: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2185/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:87
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2185/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:321
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2185/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:334
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2185/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:335
        	Error:      	"must disable replicate queue to use ChangeReplicas manually" does not contain "type of replica being removed (VOTER_FULL) does not match expectation for change"
        	Test:       	TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck
        	Messages:   	tenant=system row=0 col=2
    --- FAIL: TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck (2.24s)
=== RUN   TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_VOTERS_SkipSQLSystemTenantCheck
    multitenant_admin_function_test.go:87: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2185/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:87
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2185/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:321
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2185/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:334
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2185/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:335
        	Error:      	"must disable replicate queue to use ChangeReplicas manually" does not contain "trying to add a voter to a store that already has a VOTER_FULL"
        	Test:       	TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_VOTERS_SkipSQLSystemTenantCheck
        	Messages:   	tenant=system row=0 col=2
    --- FAIL: TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_VOTERS_SkipSQLSystemTenantCheck (2.19s)
=== RUN   TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck
    multitenant_admin_function_test.go:87: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2185/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:87
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2185/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:321
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2185/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:334
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2185/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:335
        	Error:      	"must disable replicate queue to use ChangeReplicas manually" does not contain "type of replica being removed (VOTER_FULL) does not match expectation for change"
        	Test:       	TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck
        	Messages:   	tenant=system row=0 col=2
    --- FAIL: TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck (2.23s)
=== RUN   TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_VOTERS_SkipSQLSystemTenantCheck
    multitenant_admin_function_test.go:87: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2185/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:87
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2185/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:321
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2185/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:334
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2185/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:335
        	Error:      	"must disable replicate queue to use ChangeReplicas manually" does not contain "trying to add a voter to a store that already has a VOTER_FULL"
        	Test:       	TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_VOTERS_SkipSQLSystemTenantCheck
        	Messages:   	tenant=system row=0 col=2
    --- FAIL: TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_VOTERS_SkipSQLSystemTenantCheck (2.03s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiTenantAdminFunction.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-22266

sql.TestMultiTenantAdminFunction [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7912277?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7912277?buildTab=artifacts#/) on master @ [f2b00e8039af6ea8887ec124dad8daf19da6fbf1](https://github.com/cockroachdb/cockroach/commits/f2b00e8039af6ea8887ec124dad8daf19da6fbf1):


```
=== RUN   TestMultiTenantAdminFunction
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/b287de7194e17a0386ee1775f0d198ba/logTestMultiTenantAdminFunction3072767443
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestMultiTenantAdminFunction
    multitenant_admin_function_test.go:346: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/b287de7194e17a0386ee1775f0d198ba/logTestMultiTenantAdminFunction3072767443
--- FAIL: TestMultiTenantAdminFunction (121.38s)
=== RUN   TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck
    multitenant_admin_function_test.go:87: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1386/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:87
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1386/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:321
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1386/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:334
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1386/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:335
        	Error:      	"must disable replicate queue to use ChangeReplicas manually" does not contain "type of replica being removed (VOTER_FULL) does not match expectation for change"
        	Test:       	TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck
        	Messages:   	tenant=system row=0 col=2
    --- FAIL: TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck (1.89s)
=== RUN   TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_VOTERS_SkipSQLSystemTenantCheck
    multitenant_admin_function_test.go:87: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1386/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:87
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1386/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:321
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1386/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:334
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1386/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:335
        	Error:      	"must disable replicate queue to use ChangeReplicas manually" does not contain "trying to add a voter to a store that already has a VOTER_FULL"
        	Test:       	TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_VOTERS_SkipSQLSystemTenantCheck
        	Messages:   	tenant=system row=0 col=2
    --- FAIL: TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_VOTERS_SkipSQLSystemTenantCheck (1.95s)
=== RUN   TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck
    multitenant_admin_function_test.go:87: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1386/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:87
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1386/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:321
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1386/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:334
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1386/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:335
        	Error:      	"must disable replicate queue to use ChangeReplicas manually" does not contain "type of replica being removed (VOTER_FULL) does not match expectation for change"
        	Test:       	TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck
        	Messages:   	tenant=system row=0 col=2
    --- FAIL: TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck (1.76s)
=== RUN   TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_VOTERS_SkipSQLSystemTenantCheck
    multitenant_admin_function_test.go:87: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1386/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:87
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1386/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:321
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1386/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:334
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1386/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:335
        	Error:      	"must disable replicate queue to use ChangeReplicas manually" does not contain "trying to add a voter to a store that already has a VOTER_FULL"
        	Test:       	TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_VOTERS_SkipSQLSystemTenantCheck
        	Messages:   	tenant=system row=0 col=2
    --- FAIL: TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_VOTERS_SkipSQLSystemTenantCheck (1.84s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiTenantAdminFunction.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestMultiTenantAdminFunction [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7918697?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7918697?buildTab=artifacts#/) on master @ [c050c9b4b57ecf2ceb5d449c31c617fe12c920e0](https://github.com/cockroachdb/cockroach/commits/c050c9b4b57ecf2ceb5d449c31c617fe12c920e0):


```
=== RUN   TestMultiTenantAdminFunction
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/b287de7194e17a0386ee1775f0d198ba/logTestMultiTenantAdminFunction1993874015
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestMultiTenantAdminFunction
    multitenant_admin_function_test.go:346: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/b287de7194e17a0386ee1775f0d198ba/logTestMultiTenantAdminFunction1993874015
--- FAIL: TestMultiTenantAdminFunction (147.76s)
=== RUN   TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck
    multitenant_admin_function_test.go:87: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1805/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:87
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1805/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:321
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1805/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:334
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1805/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:335
        	Error:      	"must disable replicate queue to use ChangeReplicas manually" does not contain "type of replica being removed (VOTER_FULL) does not match expectation for change"
        	Test:       	TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck
        	Messages:   	tenant=system row=0 col=2
    --- FAIL: TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck (2.24s)
=== RUN   TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_VOTERS_SkipSQLSystemTenantCheck
    multitenant_admin_function_test.go:87: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1805/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:87
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1805/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:321
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1805/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:334
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1805/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:335
        	Error:      	"must disable replicate queue to use ChangeReplicas manually" does not contain "trying to add a voter to a store that already has a VOTER_FULL"
        	Test:       	TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_VOTERS_SkipSQLSystemTenantCheck
        	Messages:   	tenant=system row=0 col=2
    --- FAIL: TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_VOTERS_SkipSQLSystemTenantCheck (2.33s)
=== RUN   TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck
    multitenant_admin_function_test.go:87: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1805/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:87
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1805/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:321
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1805/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:334
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1805/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:335
        	Error:      	"must disable replicate queue to use ChangeReplicas manually" does not contain "type of replica being removed (VOTER_FULL) does not match expectation for change"
        	Test:       	TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck
        	Messages:   	tenant=system row=0 col=2
    --- FAIL: TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck (2.31s)
=== RUN   TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_VOTERS_SkipSQLSystemTenantCheck
    multitenant_admin_function_test.go:87: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1805/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:87
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1805/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:321
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1805/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:334
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1805/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:335
        	Error:      	"must disable replicate queue to use ChangeReplicas manually" does not contain "trying to add a voter to a store that already has a VOTER_FULL"
        	Test:       	TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_VOTERS_SkipSQLSystemTenantCheck
        	Messages:   	tenant=system row=0 col=2
    --- FAIL: TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_VOTERS_SkipSQLSystemTenantCheck (2.26s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiTenantAdminFunction.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestMultiTenantAdminFunction [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7926173?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7926173?buildTab=artifacts#/) on master @ [942a4d468e9c8ad0ef45a7be33f0a326dfb19fef](https://github.com/cockroachdb/cockroach/commits/942a4d468e9c8ad0ef45a7be33f0a326dfb19fef):


```
=== RUN   TestMultiTenantAdminFunction
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/b287de7194e17a0386ee1775f0d198ba/logTestMultiTenantAdminFunction1357278115
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestMultiTenantAdminFunction
    multitenant_admin_function_test.go:346: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/b287de7194e17a0386ee1775f0d198ba/logTestMultiTenantAdminFunction1357278115
--- FAIL: TestMultiTenantAdminFunction (141.58s)
=== RUN   TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck
    multitenant_admin_function_test.go:87: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2828/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:87
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2828/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:321
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2828/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:334
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2828/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:335
        	Error:      	"must disable replicate queue to use ChangeReplicas manually" does not contain "type of replica being removed (VOTER_FULL) does not match expectation for change"
        	Test:       	TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck
        	Messages:   	tenant=system row=0 col=2
    --- FAIL: TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck (2.15s)
=== RUN   TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_VOTERS_SkipSQLSystemTenantCheck
    multitenant_admin_function_test.go:87: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2828/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:87
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2828/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:321
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2828/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:334
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2828/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:335
        	Error:      	"must disable replicate queue to use ChangeReplicas manually" does not contain "trying to add a voter to a store that already has a VOTER_FULL"
        	Test:       	TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_VOTERS_SkipSQLSystemTenantCheck
        	Messages:   	tenant=system row=0 col=2
    --- FAIL: TestMultiTenantAdminFunction/ALTER_RANGE_RELOCATE_VOTERS_SkipSQLSystemTenantCheck (2.20s)
=== RUN   TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck
    multitenant_admin_function_test.go:87: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2828/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:87
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2828/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:321
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2828/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:334
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2828/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:335
        	Error:      	"must disable replicate queue to use ChangeReplicas manually" does not contain "type of replica being removed (VOTER_FULL) does not match expectation for change"
        	Test:       	TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck
        	Messages:   	tenant=system row=0 col=2
    --- FAIL: TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_NONVOTERS_SkipSQLSystemTenantCheck (2.22s)
=== RUN   TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_VOTERS_SkipSQLSystemTenantCheck
    multitenant_admin_function_test.go:87: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2828/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:87
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2828/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:321
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2828/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:334
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2828/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/sql_test_/sql_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/multitenant_admin_function_test.go:335
        	Error:      	"must disable replicate queue to use ChangeReplicas manually" does not contain "trying to add a voter to a store that already has a VOTER_FULL"
        	Test:       	TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_VOTERS_SkipSQLSystemTenantCheck
        	Messages:   	tenant=system row=0 col=2
    --- FAIL: TestMultiTenantAdminFunction/ALTER_RANGE_x_RELOCATE_VOTERS_SkipSQLSystemTenantCheck (2.23s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiTenantAdminFunction.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/multitenant_admin_function_test.go
+++ b/pkg/sql/multitenant_admin_function_test.go
@@ -8,7 +8,7 @@
 // by the Apache License, Version 2.0, included in the file
 // licenses/APL.txt.
 
-package sql
+package sql_test
 
 import (
 	"context"
@@ -16,6 +16,8 @@ import (
 	"testing"
 
 	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver"
+	"github.com/cockroachdb/cockroach/pkg/sql"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
 	"github.com/cockroachdb/cockroach/pkg/util/errorutil"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
@@ -27,7 +29,15 @@ const createTable = "CREATE TABLE t(i int PRIMARY KEY);"
 const rangeErrorMessage = "RangeIterator failed to seek"
 
 func createTestServer(t *testing.T) (serverutils.TestServerInterface, func()) {
-	testCluster := serverutils.StartNewTestCluster(t, 1, base.TestClusterArgs{})
+	testCluster := serverutils.StartNewTestCluster(t, 1, base.TestClusterArgs{
+		ServerArgs: base.TestServerArgs{
+			Knobs: base.TestingKnobs{
+				Store: &kvserver.StoreTestingKnobs{
+					AllowUnsynchronizedReplicationChanges: true,
+				},
+			},
+		},
+	})
 	return testCluster.Server(0), func() {
 		testCluster.Stopper().Stop(context.Background())
 	}
@@ -52,7 +62,7 @@ func createSecondaryTenantDB(
 	_, db := serverutils.StartTenant(
 		t, testServer, base.TestTenantArgs{
 			TestingKnobs: base.TestingKnobs{
-				TenantTestingKnobs: &TenantTestingKnobs{
+				TenantTestingKnobs: &sql.TenantTestingKnobs{
 					AllowSplitAndScatter:      allowSplitAndScatter,
 					SkipSQLSystemTentantCheck: skipSQLSystemTenantCheck,
 				},

